packages/components/ContextualMenu/src/SubmenuItem.tsx

```diff
@@ -42,58 +42,87 @@ export const SubmenuItem = compose<SubmenuItemType>({
     const cmRef = useViewCommandFocus(componentRef);
 
     const onItemHoverIn = React.useCallback((e) => {
-      userProps.onHoverIn(e);
+      componentRef.current.focus();
+      userProps.onHoverIn && userProps.onHoverIn(e);
     }, []);
 
-    const onItemClick = React.useCallback(
+    const onHoverInDelay = Platform.select({
+      macos: 100,
+      default: 500, // win32
+    });
+
+    const onItemHoverOut = React.useCallback((e) => {
+      context?.dismissSubmenu && context.dismissSubmenu();
+      userProps.onHoverOut && userProps.onHoverOut(e);
+    }, []);
+
+    const onItemPress = React.useCallback(() => {
+      if (!disabled) {
+        onClick && onClick();
+        context?.onItemClick && context.onItemClick(itemKey);
+        context?.dismissSubmenu && context.dismissSubmenu();
+        context?.onDismissMenu();
+      }
+    }, [context, disabled, itemKey, onClick]);
+
+    const pressable = useAsPressable({
+      ...rest,
+      onPress: onItemPress,
+      onHoverIn: onItemHoverIn,
+      delayHoverIn: onHoverInDelay,
+      onHoverOut: onItemHoverOut,
+    });
+
+    /**
+     * GH #blah:
```

Comment:
Should there be a number for this?

Reply:
Ah, good catch. I put blah as an obvious-for-others placeholder if I forgot to update it with a real number